libs/wingc/src/parser.rs

```diff
@@ -735,54 +801,95 @@ impl<'s> Parser<'s> {
 			None
 		};
 
-		// if the module name is a path ending in .w, create a new Parser to parse it as a new Scope,
-		// and create a StmtKind::Module instead
-		if module_name.name.starts_with("\"") && module_name.name.ends_with(".w\"") {
-			let module_path = Utf8Path::new(&module_name.name[1..module_name.name.len() - 1]);
+		let module_path = Utf8Path::new(&module_name.name[1..module_name.name.len() - 1]);
+		if is_absolute_path(&module_path) {
+			return self.with_error(
+				format!("Cannot bring \"{}\" since it is not a relative path", module_path),
+				statement_node,
+			);
+		}
+
+		if module_name.name.starts_with("\".") && module_name.name.ends_with("\"") {
 			let source_path = normalize_path(module_path, Some(&Utf8Path::new(&self.source_name)));
 			if source_path == Utf8Path::new(&self.source_name) {
 				return self.with_error("Cannot bring a module into itself", statement_node);
 			}
 			if !source_path.exists() {
-				return self.with_error(format!("Cannot find module \"{}\"", source_path), statement_node);
+				return self.with_error(format!("Cannot find module \"{}\"", module_path), statement_node);
 			}
-			if !source_path.is_file() {
-				return self.with_error(
-					format!("Cannot bring module \"{}\": not a file", source_path),
-					statement_node,
-				);
-			}
-			if source_path.ends_with(".main.w\"") || source_path.ends_with(".test.w\"") {
+
+			// case: .w file
+			if source_path.ends_with("/main.w") || source_path.ends_with(".main.w\"") || source_path.ends_with(".test.w\"") {
```

Comment:
maybe:
```suggestion
			if source_path.ends_with("/main.w\"") || source_path.ends_with(".main.w\"") || source_path.ends_with(".test.w\"") {
```
Also why fail and not just silently ignore them. This error won't let you bring something that also has an entry point. Finally what about "test.w" files? Should we ignore them too?

Reply:
Fixed (turns out we already have a helper function named `is_entrypoint_file` for this)

> Also why fail and not just silently ignore them. This error won't let you bring something that also has an entry point.

I'm not sure I'm understanding your suggestion. Basically this rule / error is enforcing that there are two types of Wing files, entrypoint files and module files, and you can only "bring" other module files. What's the behavior you're proposing?

I've added another case to `bring_local_self.test.w` that shows the error message since it looks like I missed this one

> Finally what about "test.w" files? 

I don't have a strong opinion on it, but I'd rather leave for another PR if it's ok